api/src/main/resources/org/openmrs/module/mirebalaisreports/sql/mexicoCesMeds.sql

```diff
@@ -0,0 +1,37 @@
+SELECT
+    MAX(d.name) "Medicamento",
+    SUM(disp_quant.value_numeric) "Cantidad"
+FROM obs disp_quant
+         LEFT JOIN obs disp_drug
+                   ON disp_drug.obs_group_id = disp_quant.obs_group_id
+                       AND disp_drug.concept_id = (
```

Comment:
I'd really like to turn the code in lines 8-17 and again on lines 22-30 into a reusable function, and then to use that function in both of these spots as well as for all of the other reports out there that do something similar.  If this doesn't perform well, then consider creating a temporary table containing concept_id, source_name or hl7_code, term that can easily be joined against.
The function I'm thinking would basically use the logic here, but be wrapped up like:

```
DROP FUNCTION IF EXISTS concept_by_mapping;

CREATE FUNCTION concept_by_mapping(_sourceName varchar(50), _term varchar(50))
    RETURNS INT
    DETERMINISTIC
BEGIN
    DECLARE ret INT;

    SELECT concept_id into ret
        FROM ...

    return ret;
END
```
I've done this in the malawi project and can show you how I set these up.  @dmdesimone may have done something like this already in mirebalaisreporting, not sure...

Reply:
Absolutely, yes.